src/types/types.ts

```diff
@@ -1166,6 +1166,205 @@ export interface NewApiKeyResponse extends ApiKeyObject {
   value: string;
 }
 
+// groups
+
+export interface ListGroups {
+  /**
+   * Whether to get all groups, only available with the groups:list acl.
+   */
+  all?: boolean;
+}
+
+/**
+ * Name of the group
+ * @example Production Engineers
+ */
+export type GroupName = string;
+
+/**
+ * ID of the group in the source. If this is the same ID as a group in the IDP, a user in that group will implicitly be a part of this group as well.
+ * @example b7c9a5a4-99c2-4785-bed3-5e6ad9a78603
+ */
+export type GroupSourceId = string;
+
+export interface Group {
+  name: GroupName;
+  sourceId?: GroupSourceId;
+  capabilities?: cogniteCapability;
+  id: number;
+  isDeleted: boolean;
+  deletedTime?: Date;
+}
+
+export type LIST = 'LIST';
+export type READ = 'READ';
+export type WRITE = 'WRITE';
+export type CREATE = 'CREATE';
+export type UPDATE = 'UPDATE';
+export type DELETE = 'DELETE';
+export type MEMBEROF = 'MEMBEROF';
+export type EXECUTE = 'EXECUTE';
+
+export interface Acl<ActionsType, ScopeType> {
+  actions: ActionsType[];
+  scope: ScopeType;
+}
+
+export interface GenericAclAllScope {
+  all: {};
+}
+
+export interface GenericAclCurrentUserScope {
+  currentuserscope: {};
+}
+
+export type cognitegroups_aclAction = LIST | READ | CREATE | UPDATE | DELETE;
+export type cognitegroups_aclScope =
+  | GenericAclAllScope
+  | GenericAclCurrentUserScope;
+export type cognitegroups_aclAcl = Acl<
```

Comment:
the naming is really bad.. It's like reading C code
what does `aclAcl` mean? underscore?